src/fsharp/MethodOverrides.fs

```diff
@@ -420,7 +421,83 @@ module DispatchSlotChecking =
                     | meth :: _ when meth.IsFinal -> errorR(Error(FSComp.SR.tcCannotOverrideSealedMethod((sprintf "%s::%s" (meth.ApparentEnclosingType.ToString()) (meth.LogicalName))), m))
                     | _ -> ()
 
-
+    let GetInterfaceOverrideByRequiredSlotsFromTypes g amap m infoReader reqdTy impliedTys reqdSlot =
+        if isInterfaceTy g reqdTy then
+            impliedTys
+            |> List.choose (fun ty ->
+                if isInterfaceTy g ty && TypeFeasiblySubsumesType 0 g amap m reqdTy CanCoerce ty then
+                    match TryFindILIntrinisicOverrideByMethInfo infoReader m AccessibleFromSomewhere ty reqdSlot with
+                    | Some minfo -> Some (ty, RequiredSlot (reqdSlot, not minfo.IsAbstract))
+                    | _ -> None
+                else
+                    None)
+        else
+            []
+
+    let GetDispatchSlots g amap reqdTyRange infoReader denv availImpliedInterfaces reqdTyInfos reqdTy impliedTys =
+        let dispatchSlots = 
+             [ if isInterfaceTy g reqdTy then 
+                   for impliedTy in impliedTys  do
+                       // Check if the interface has an inherited implementation
+                       // If so, you do not have to implement all the methods - each
+                       // specific method is "optionally" implemented.
+                       let isOptional = 
+                           ListSet.contains (typeEquiv g) impliedTy availImpliedInterfaces
+                       for reqdSlot in GetImmediateIntrinsicMethInfosOfType (None, AccessibleFromSomewhere) g amap reqdTyRange impliedTy do
+                         // Interface methods that are overriden in another interface will always be final, even when the method is re-abstracted.
+                         // We do not want to look at the methods that override.
+                         if not reqdSlot.IsFinal then
+                             if isOptional || not reqdSlot.IsILMethod then //|| not (g.langVersion.SupportsFeature LanguageFeature.DefaultInterfaceMethodsInterop) then
```

Comment:
Is this commented out check going to go in, or can it be removed?

Reply:
I have it temporarily commented. This part is still WIP.